lib/providers/user_info_provider.dart

```diff
@@ -56,4 +57,25 @@ class UserInfo extends _$UserInfo {
       return _saveEncoded(user);
     });
   }
+  // TODO: need to fix this
+  Future<void> signUpWithRole(MemberRole role) async {
+    state = await AsyncValue.guard(() async {
+      final authClient = await ref.read(authClientProvider.future);
+      final member = await authClient.createMember(
+        data: MemberCreateBody(role: role),
+      );
+      await ref.read(accessTokenProvider.notifier).setValue(member.uuid);
+      final user = User(role: role, uuid: member.uuid);
+      return _saveEncoded(user);
+    });
+  }
+  
+  Future<void> logout() async {
+    state = await AsyncValue.guard(() async {
+      final prefs = ref.read(prefsProvider);
+      await prefs.remove(_key);
+      await ref.read(accessTokenProvider.notifier).clear();
+      return User.defaultValue;
+    });
+  }
```

Comment:
로그아웃 하더라도 `role`은 유지되어야 할 것 같습니다.

```suggestion
  Future<void> logout() async {
    state = await AsyncValue.guard(() async {
      final data = await future;
      /* ... */
      final user = data.copyWith(uuid: null);
      return _saveEncoded(user);
    });
  }
```

Reply:
이유가 있을까요? userInfo의 role을 watch하고 있는 프로바이더가 있다거나 이런 이유일까요?

Reply:
네 맞습니다. 하지만 그것 보다도 `RoleSelectionPage`에서 이미 설정된 `role`이기 때문에 단순히 로그아웃한다고 해서 변경되면 안된다고 생각합니다.